CHANGELOG.md

```diff
@@ -2,6 +2,128 @@
 
 All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.
 
+## [2.0]() (UNRELEASED)
+
+Version 2.0 is a bigger change with the main goal to make this library more reliable and future safe.
+See https://github.com/andygrunwald/go-jira/issues/489 for details.
+
+### Migration
+
+#### Split of clients
+
+We moved from 1 client that handles On-Premise and Cloud to 2 clients that handle either On-Premise or Cloud.
+Previously you used this library like:
+
+```go
+import (
+    "github.com/andygrunwald/go-jira"
+)
+```
+
+In the new version, you need to decide if you interact with the Jira On-Premise or Jira Cloud version.
+For the cloud version, you will import this library like
+
+```go
+import (
+	jira "github.com/andygrunwald/go-jira/cloud"
+)
+```
+
+For On-Premise it looks like
+
+```go
+import (
+	jira "github.com/andygrunwald/go-jira/onpremise"
+)
+```
+
+#### Init a new client
+
+The order of arguments in the `jira.NewClient` has changed:
+
+1. The base URL of your JIRA instance
+2. A HTTP client (optional)
+
+Before:
+
+```go
+jira.NewClient(nil, "https://issues.apache.org/jira/")
+```
+
+After:
+
+```go
+jira.NewClient("https://issues.apache.org/jira/", nil)
+```
+
+#### User Agent
+
+The client will identify itself via a UserAgent `go-jira/2.0.0`.
+
+#### `NewRawRequestWithContext` removed, `NewRawRequest` requires `context`
+
+The function `client.NewRawRequestWithContext()` has been removed.
+`client.NewRawRequest()` accepts now a context as the first argument.
+This is a drop in replacement.
+
+Before:
+
+```go
+client.NewRawRequestWithContext(context.Background(), "GET", .....)
```

Comment:
```suggestion
client.NewRawRequestWithContext(context.Background(), http.MethodGet, .....)
```
The http constants should be used instead of using a string.

Reply:
Totally right. I made a issue to cover this globally, see https://github.com/andygrunwald/go-jira/issues/508